src/services/refactors/generateGetAccessorAndSetAccessor.ts

```diff
@@ -119,7 +119,12 @@ namespace ts.refactor.generateGetAccessorAndSetAccessor {
 
     function getConvertibleFieldAtPosition(file: SourceFile, startPosition: number): Info | undefined {
         const node = getTokenAtPosition(file, startPosition, /*includeJsDocComment*/ false);
-        const declaration = findAncestor(node.parent, isAcceptedDeclaration);
+        const declaration = <AcceptedDeclaration>findAncestor(node.parent, n => {
+            if (isFunctionLikeDeclaration(n)) {

```

Comment:
it is still wired that i can select the value, and get the refactoring.. e.g.:

```ts
class C {
    prop = [  1, 2 , 3  ]; // select 2
}
```

I think the better approach here is to make sure whatever declaration we found, its name is part of the original span..